src/components/LinkModal/LinkModal.js

```diff
@@ -0,0 +1,253 @@
+        link.Author = core.getCurrentUser();
+        link.Subject = 'Link';
+        link.X = Math.min(quad.x1, quad.x3);
+        link.Y = Math.min(quad.y1, quad.y3);
+        link.Width = Math.abs(quad.x1 - quad.x3);
+        link.Height = Math.abs(quad.y1 - quad.y3);
+        linksResults.push(link);
+      });
+      createHighlightAnnot(linksResults, currPageQuads, selectedText);
+    }
+
+    if (selectedAnnotations && selectedAnnotations.length === 1) {
+      const link = new Annotations.Link();
+      link.PageNumber = currentPage;
+      link.StrokeColor = new Annotations.Color(0, 165, 228);
+      link.StrokeStyle = 'underline';
+      link.StrokeThickness = 2;
+      link.Author = core.getCurrentUser();
+      link.Subject = 'Link';
+      link.X = selectedAnnotations[0].X;
+      link.Y = selectedAnnotations[0].Y;
+      link.Width = selectedAnnotations[0].Width;
+      link.Height = selectedAnnotations[0].Height;
+      linksResults.push(link);
+      selectedAnnotations[0].associateLink([link.Id]);
+    }
+
+    return linksResults;
+  };
+
+  const createHighlightAnnot = async(linkAnnotArray, quads, text) => {
+    const linkAnnot = linkAnnotArray[0];
+    const highlight = new Annotations.TextHighlightAnnotation();
+    highlight.PageNumber = linkAnnot.PageNumber;
+    highlight.X = linkAnnot.X;
+    highlight.Y = linkAnnot.Y;
+    highlight.Width = linkAnnot.Width;
+    highlight.Height = linkAnnot.Height;
+    highlight.StrokeColor = new Annotations.Color(0, 0, 0, 0);
+    highlight.Quads = quads;
+    highlight.Author = core.getCurrentUser();
+    highlight.setContents(text);
+
+    let linkAnnotIdArray = [];
+    linkAnnotArray.forEach(link => {
+      linkAnnotIdArray.push(link.Id);
+    });
+    highlight.associateLink(linkAnnotIdArray);
+
+    await core.addAnnotations([highlight]);
+    await core.drawAnnotationsFromList([highlight]);
+  };
+
+  const addURLLink = () => {
+    const links = createLink();
+
+    const action = new window.Actions.URI({ uri: url });
+    links.forEach(async link => {
+      link.addAction('U', action);
+      await core.addAnnotations([link]);
+      await core.drawAnnotationsFromList([link]);
+    });
+
+    closeModal();
+  };
+
+  const addPageLink = () => {
+    const links = createLink();
+
+    const Dest = window.Actions.GoTo.Dest;
+    const options = { dest: new Dest({ page: pageNumber }) };
+    const action = new window.Actions.GoTo(options);
+
+    links.forEach(async link => {
+      link.addAction('U', action);
+      await core.addAnnotations([link]);
+      await core.drawAnnotationsFromList([link]);
+    });
+
+    closeModal();
+  };
+
+  useEffect(() => {
+    if (isOpen) {
+      //  prepopulate page numbers for the dropdown
+      const numbers = [];
+      for (let i = 1; i <= totalPages; i++) {
+        numbers.push(
+          <option key={i} value={i}>
+            {i}
+          </option>,
+        );
+      }
+      setDropdownNumbers(numbers);
```

Comment:
Instead of storing `<option>`s on a state can we just have them in the returned JSX?
```jsx
                <select
                  className="pageNumberSelect"
                  ref={pageNumberInput}
                  value={pageNumber}
                  onChange={e => setPageNumber(e.target.value)}
                >
                  {/* create <option>s here based on the value of totalPages */}
                </select>
```

Reply:
Fixed that